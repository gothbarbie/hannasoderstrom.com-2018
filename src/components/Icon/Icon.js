// @flow
import React from 'react'
import styled from 'styled-components'

const IconStyle = styled.span`
  color: ${({ color }) => color};
  font-size: ${({ size }) => size && size + 'rem'};
  font-size: 2rem;
`

type Props = {
  name: string,
  size?: string,
  color?: string,
}

const Icon = ({ name, size, color }: Props) => (
  <IconStyle color={color} size={size}>
    <i className={`fas fa-${name}`} />
  </IconStyle>
)

export default Icon
