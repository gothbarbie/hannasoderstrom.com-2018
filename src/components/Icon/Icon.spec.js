import React from 'react'
import { shallow } from 'enzyme'

import Icon from './index'

describe('components/Icon', () => {
  let component

  beforeEach(() => {
    component = shallow(<Icon />)
  })

  it('renders', () => {
    expect(component).toMatchSnapshot()
  })
})
