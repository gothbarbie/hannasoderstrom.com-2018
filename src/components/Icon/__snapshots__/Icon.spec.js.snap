// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Icon renders 1`] = `
<styled.span>
  <i
    className="fas fa-undefined"
  />
</styled.span>
`;
